backend/src/server/config/rateLimiter.ts

```diff
@@ -74,3 +62,38 @@ export const publicEndpointLimit: RateLimitOptions = {
   max: 30,
   keyGenerator: (req) => req.realIp
 };
+
+async function fetchRateLimitsFromDb(db: Knex) {
+  try {
+    const rateLimitDAL = rateLimitDALFactory(db);
```

Comment:
This pattern feels very intuitive. I would recommend instead load the data from the  outside and just pass the configuration thus avoiding all these in between things.